packages/popover/test/trigger.test.js

```diff
@@ -0,0 +1,390 @@
+      focusin(target);
+      await nextRender();
+
+      mouseenter(target);
+      focusout(target);
+      await nextUpdate(popover);
+      expect(overlay.opened).to.be.true;
+    });
+
+    it('should close on target focusout if target has lost hover', async () => {
+      focusin(target);
+      await nextRender();
+
+      mouseenter(target);
+      mouseleave(target);
+      focusout(target);
+      await nextUpdate(popover);
+      expect(overlay.opened).to.be.false;
+    });
+
+    it('should not close on target focusout if overlay has hover', async () => {
+      focusin(target);
+      await nextRender();
+
+      mouseenter(overlay);
+      focusout(target);
+      await nextUpdate(popover);
+      expect(overlay.opened).to.be.true;
+    });
+
+    it('should close on target focusout if overlay has lost hover', async () => {
+      focusin(target);
+      await nextRender();
+
+      mouseenter(overlay);
+      mouseleave(overlay);
+      focusout(target);
+      await nextUpdate(popover);
+      expect(overlay.opened).to.be.false;
+    });
+
+    it('should not close on target mouseleave if target has focus', async () => {
+      mouseenter(target);
+      await nextRender();
+
+      focusin(target);
+      mouseleave(target);
+      await nextUpdate(popover);
+      expect(overlay.opened).to.be.true;
+    });
+
+    it('should not close on target mouseleave if overlay has focus', async () => {
+      mouseenter(target);
+      await nextRender();
+
+      focusin(overlay);
+      mouseleave(target);
+      await nextUpdate(popover);
+      expect(overlay.opened).to.be.true;
+    });
+
+    it('should not close on overlay mouseleave if overlay has focus', async () => {
+      mouseenter(target);
+      await nextRender();
+
+      focusin(overlay);
+      mouseenter(overlay);
+      mouseleave(overlay);
+      await nextUpdate(popover);
+      expect(overlay.opened).to.be.true;
+    });
+  });
+
+  describe('manual', () => {
+    [null, undefined, []].forEach((value) => {
+      const trigger = Array.isArray(value) ? 'empty array' : value;
+
+      describe(`trigger set to ${trigger}`, () => {
+        beforeEach(async () => {
+          popover.trigger = value;
+          await nextUpdate(popover);
+        });
+
+        it(`should set restoreFocusOnClose to false with trigger set to ${trigger}`, () => {
+          expect(overlay.restoreFocusOnClose).to.be.false;
+        });
+
+        it(`should not open on target click with trigger set to ${trigger}`, async () => {
+          target.click();
+          await nextRender();
+          expect(overlay.opened).to.be.false;
+        });
+
+        it(`should not open on target mouseenter with trigger set to ${trigger}`, async () => {
+          mouseenter(target);
+          await nextRender();
+          expect(overlay.opened).to.be.false;
+        });
+
+        it(`should not open on target focusin with trigger set to ${trigger}`, async () => {
```

Comment:
It would be great to also test that the popover doesn't close on focusout and mouseleave

Reply:
Done.